addons/account/models/account_move.py

```diff
@@ -3531,7 +3531,9 @@ def _get_update_invoice_from_attachment_decoders(self, invoice):
     def _compute_narration(self):
         use_invoice_terms = self.env['ir.config_parameter'].sudo().get_param('account.use_invoice_terms')
         for move in self:
-            if not use_invoice_terms or not move.is_sale_document(include_receipts=True):
+            if not move.is_sale_document(include_receipts=True):
```

Comment:
```suggestion
            if not move.is_invoice(include_receipts=True):
```
? :thinking:

Reply:
Also,
- you can test this field directly in `account` in the bank statement file instead of calling the bank reco widget
- you probably have the same issue in 14.0

Reply:
`is_sale_document` instead of ' is_invoice` because field `narration` is use for terms & conditions, only in customer invoices/refunds.